lib/process-services/src/lib/task-list/components/task-filters/task-filters.component.ts

```diff
@@ -136,10 +137,20 @@ export class TaskFiltersComponent implements OnInit, OnChanges {
                 if (res.length === 0 && this.isFilterListEmpty()) {
                     this.createFiltersByAppId(appId);
                 } else {
-                    this.resetFilter();
-                    this.filters = res;
-                    this.selectFilter(this.filterParam);
-                    this.success.emit(res);
+                    const migratedFilters = this.migrateObsoleteFilters(res);
+                    if (migratedFilters.length > 0) {
+                        forkJoin(migratedFilters).subscribe(() => {
+                            this.resetFilter();
+                            this.filters = res;
+                            this.selectFilter(this.filterParam);
+                            this.success.emit(res);
+                        });
+                    } else {
+                        this.resetFilter();
```

Comment:
This code is the same like code inside subscribe above - can we move it to some reusable function?

Reply:
Moved to separate function